fluentinterface/src/main/java/com/iluwatar/fluentinterface/fluentiterable/simple/SimpleFluentIterable.java

```diff
@@ -0,0 +1,194 @@
+package com.iluwatar.fluentinterface.fluentiterable.simple;
+
+import com.iluwatar.fluentinterface.fluentiterable.FluentIterable;
+
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * This is a simple implementation of the FluentIterable interface. It evaluates
+ * all chained operations eagerly.
+ * @param <TYPE> the type of the objects the iteration is about
+ */
+public class SimpleFluentIterable<TYPE> implements FluentIterable<TYPE> {
+
+    private final Iterable<TYPE> iterable;
+
+    /**
+     * This constructor creates a copy of a given iterable's contents.
+     * @param iterable the iterable this interface copies to work on.
+     */
+    protected SimpleFluentIterable(Iterable<TYPE> iterable) {
```

Comment:
Feedback
 - I got it. We can have `protected` constructors.
 - I think we should use incoming `Iterable` itself because I would like if the original collection is altered so that I don't have to pay extra cost. In case user does not want to mutate original collection then we can provide other static factory method `fromCopyOf(Iterable)`. Then it is upto the user. I am leaning towards mutating original because that's what `LazyFluentIterable` does, so it becomes homogeneous :smiley: